app/controllers/rentals_controller.rb

```diff
@@ -0,0 +1,73 @@
+class RentalsController < ApplicationController
+  before_action :require_customer
+  before_action :require_video
+
+  def create
+    rental = Rental.new(customer_id: @customer.id, video_id: @video.id)
+
+    if rental.video.available_inventory == 0
+      render json: { errors: ["No available copies of the video available"] }, status: :bad_request
+      return
+    end
+
+    if rental.save
+      rental.customer.videos_checked_out_count += 1
+      rental.customer.save
+      rental.video.available_inventory -= 1
+      rental.video.save
+      
+      render json: {
+        customer_id: rental.customer_id,
+        video_id: rental.video_id,
+        due_date: rental.due_date,
+        videos_checked_out_count: rental.customer.videos_checked_out_count,
+        available_inventory: rental.video.available_inventory
+      }, status: :ok
```

Comment:
Most of this logic should have been encapsulated into a model method to handle checking out.  

```suggestion
    if video.available_inventory == 0
      render json: { errors: ["No available copies of the video available"] }, status: :bad_request
      return
    end
    
    rental = Rental.new(customer: @customer, video: @video)
    
    if rental.check_out
      render json: {
        customer_id: rental.customer_id,
        video_id: rental.video_id,
        due_date: rental.due_date,
        videos_checked_out_count: rental.customer.videos_checked_out_count,
        available_inventory: rental.video.available_inventory
      }, status: :ok
```

```ruby
class Rental < ApplicationRecord
  # Relationships, validations, etc...

  def check_out(customer, video)
    if rental.save
      rental.customer.videos_checked_out_count += 1
      rental.customer.save
      rental.video.available_inventory -= 1
      rental.video.save
      
      return rental
    else
      return nil
    end
  end
end
```

Reply:
In a production application you would also probably want to use a [transaction](https://api.rubyonrails.org/classes/ActiveRecord/Transactions/ClassMethods.html) for safety when updating multiple tables like this, though we haven't taught you those (and don't as part of the curriculum due to a lack of time).